packages/core/types/index.js

```diff
@@ -474,6 +477,15 @@ export type DependencyOptions = {|
    * @default 'sync'
    */
   +priority?: DependencyPriority,
+  /**
+   * Controls the behavior of the bundle the resolved asset is placed into. Use in combination with `priority`
+   * to determine when the bundle is loaded.
+   *   - inline: The resolved asset will be placed into a new inline bundle. Inline bundles are not written
+   *       to a separate file, but embedded into the parent bundle.
+   *   - isolated: The resolved asset will be isolated from its parents in a separate bundle.
+   *       Shared assets will be duplicated.
+   */
+  +bundleBehavior?: BundleBehavior,
```

Comment:
API wise I'm not too sure about this. I considered combining bundleBehavior and priority, but I think it's possible to have an isolated bundle that is either parallel or lazy (but probably not sync), and it's possible to have an inline sync or async (though parallel vs lazy doesn't really matter). 🤔